python/ray/train/v2/_internal/execution/worker_group/worker_group.py

```diff
@@ -778,3 +781,18 @@ def get_lowest_gpu_id(worker) -> int:
         for workers in node_id_to_workers.values():
             sorted_workers.extend(workers)
         return sorted_workers
+
+    @staticmethod
+    def _get_worker_runtime_env(custom_runtime_env: Dict) -> Dict:
+        """Update custom runtime env with internal Ray Train env vars
+        that should be propagated from the driver to worker processes.
+
+        Args:
+            The custom runtime env dict passed in by the user.
+        """
+        merged_env_vars = get_env_vars_to_propagate()
+        merged_env_vars.update(custom_runtime_env.get("env_vars", {}))
+
+        runtime_env = dict(custom_runtime_env)
```

Comment:
nit: I don't think you need this explicit casting since RuntimeEnvironment subclasses `dict`.

Reply:
Oh, I wanted to make a copy of the dict regardless so that the user's runconfig doesn't get modified in place.

I think this technically gets run on the train controller, not the driver, so it's not a big concern.

But that's true I need to make it clear that a RuntimeEnv type could be passed in here.